src/button-group/menu-dropdown-item.tsx

```diff
@@ -26,7 +26,7 @@ const MenuDropdownItem = React.forwardRef(
   ) => {
     const containerRef = React.useRef<HTMLDivElement>(null);
     const onClickHandler = (event: CustomEvent<ButtonDropdownProps.ItemClickDetails>) => {
-      fireCancelableEvent(onItemClick, { id: event.detail.id }, event);
+      fireCancelableEvent(onItemClick, { id: event.detail.id, checked: event.detail.checked }, event);
```

Comment:
Adding this to support selectable button dropdown items: https://cloudscape.design/components/button-dropdown?tabId=playground&example=with-selectable-items